### DISCUSSION TITLE
Error "Collection XXX doesn't exist" when creating a M2A relationship

### DISCUSSION THREAD

**Discussion by powerseed** [directus/directus#10244] [Archive]
I tried to create a M2A relationship, but after the last step the following error showed up

![image](https://user-images.githubusercontent.com/42278237/144509571-2a9b7bd6-12a6-4ca5-b303-b23526b453b4.png)

The collection I created the M2A relationship in is called `Film`, and `assets` is the name I gave the key, as shown below.

![image](https://user-images.githubusercontent.com/42278237/144510521-a28cf67c-4369-45fc-88cf-a3df8a9dd013.png)

I am wondering why the error.

I mean, of course the collection `film_assets` doesn't exist, because I didn't create it and wasn't referencing it. 

The junction table I chose in the `Relationship` tab is another table.

My Directus version is `9.0.0 rc83`

Hope someone can help out.

Thanks!

**Comment by rijkvanzanten** [comment]
> My Directus version is 9.0.0 rc83

I would strongly suggest upgrading to the latest version if at all possible, or at least 9.0.0 or up! The relations engine has been reworked drastically since rc.83.

**Reply by powerseed** [reply]
@rijkvanzanten 
Thanks for the hint. I updated to the latest version and the error has disappeared.

But I am still quite confused when creating M2A relationship.

1. I am wondering what is the field highlighted in the following screenshot supposed to be?

![image](https://user-images.githubusercontent.com/42278237/144538525-59e477c3-f588-4a32-ba20-6a9641bfe6f5.png)

I assume it should be a column in the junction table that is a foreign key referencing another table.

If yes, shouldn't I be able to choose this column from a dropdown? The reality is that I am asked to enter it manually. 

If no, I have no clue about what should I enter in this field.

2. According to the official doc, as shown below, each table should have an alias field. 

![image](https://user-images.githubusercontent.com/42278237/144539057-2d91c61c-2a7f-4a6f-aaf6-8a400e30ae32.png)

I am wondering if I need to create each of them manually or are they created automatically by the M2A relationship?

Thanks!

**Reply by rijkvanzanten** [reply]
Heya! In your screenshot, the junction table holds the following fields:

- `film_id`: Primary key of the parent item, in this case from `film`
- `collection`: The name of the related collection. In your particular setup, you only allow the user to choose from a single related collection, so the value will always be `video_asset`
- `item`: The primary key of the related item in the related collection.

> If yes, shouldn't I be able to choose this column from a dropdown?

Yes, when your junction collection already exists, this field should also have the dropdown picker on the side. I'll look into that little bug now 👍🏻 

> If no, I have no clue about what should I enter in this field.

You can enter any name you want. You don't have to manually create anything. If you fill in a name of a column that doesn't exist yet, Directus will auto-generate it for you.

> According to the official doc, as shown below, each table should have an alias field.

That particular page is more a reference to outline what's going on under-the-hood. It does not mean you have to set all of those up manually 🙂 I'll add a little notice tip box to that page to highlight this 👍🏻

**Reply by powerseed** [reply]
@rijkvanzanten 

Thanks for the detailed reply!

Since you mentioned that I don't need to create the alias fields manually, I just gave it try, and encountered the following error after clicking a row in the `film` collection (where I created the M2A relationship in)

![image](https://user-images.githubusercontent.com/42278237/144647969-b5b8983e-6bd6-46a5-86d6-d420b8961817.png)

And the M2A field is like shown below, where the `Create New` and `Add Existing` dropdowns are both empty.

![image](https://user-images.githubusercontent.com/42278237/144648301-957edd8f-9806-44d9-8c97-1808ced55624.png)

Since the error message doesn't really indicate where went wrong, could you provide me with a little bit of hints please?

Thanks again!

**Reply by rijkvanzanten** [reply]
Hmm hard to tell without seeing the full schema and all relationships. Could you try removing the junction table you created manually before, and have Directus auto-generate it?

Also, are you only linking to video_assets, or are you intending on adding more types later on? If you're only relating to items of a single type, I'd recommend going with a regular M2M instead 🙂

**Reply by powerseed** [reply]
@rijkvanzanten 

Thanks for the hints!

Using an auto-generated junction table works!

So, I assume the problem lies in the `collection` field of my existing junction table, where the field is empty as shown below

![image](https://user-images.githubusercontent.com/42278237/144660424-62f18ba6-a12a-422e-8afe-33c79650ddb6.png)

I am wondering if I can enter values into this field? It seems not allowed.

If it is allowed, what kind of values should I enter in my case?

Since my table's name is `video_asset`, should the `collection` field have the same value (`video_asset`)?

> If you're only relating to items of a single type, I'd recommend going with a regular M2M instead

My plan is to relate multiple types. My example shown above is just for simplicity. I would like to make sure that it works first, then I will add more types into the relationship.

Thank you!

**Reply by rijkvanzanten** [reply]
> I am wondering if I can enter values into this field? It seems not allowed.

You can, and it's allowed. It could be that the field is set to readonly / hidden though.

> Since my table's name is video_asset, should the collection field have the same value (video_asset)?

That's correct!

**Reply by powerseed** [reply]
@rijkvanzanten 

I feel like I am very close to success.

After filling the `collection` field of the junction table, the error `[UNKNOWN] Cannot read properties of undefined (reading 'meta')` still shows up. I logged out and logged in, but that didn't solved the problem (sometimes it does).

Below is how I configured the M2A relationship:

![image](https://user-images.githubusercontent.com/42278237/144662866-f28b6293-870d-402b-899b-2259a2e35801.png)

And below is what my junction table looks like:

![image](https://user-images.githubusercontent.com/42278237/144663096-db5695a5-1d25-4ae9-91d7-9cf8c9249472.png)

Please help! I have the feeling that this is the very last issue I need to solve.

**Reply by rijkvanzanten** [reply]
Where/when does that error occur?

**Reply by powerseed** [reply]
@rijkvanzanten 

After I click on a row of table `film`

![image](https://user-images.githubusercontent.com/42278237/144664021-aa804979-a2fb-4eb4-9e81-bd2911b3a2a7.png)

The view for one single record still opens, but in the center of the screen there is an error dialog.

And the M2A field's two dropdowns are both empty as shown below.

![image](https://user-images.githubusercontent.com/42278237/144664336-e63a4635-807e-48c1-b148-4dd13942ca75.png)

**Reply by powerseed** [reply]
@rijkvanzanten 

It works!

It was my bad.

I manually inserted a record for the junction table I used for the M2A relationship.

That record is for a M2M one. I forgot to remove it and Directus didn't overwrite it, so the M2A doesn't work.

I am so sorry for wasting your time because of this.

I really appreciate your help!

**Reply by rijkvanzanten** [reply]
Glad you figured it out!

> I am so sorry for wasting your time because of this.

No worries! Always important to discuss to make sure it isn't a bug in Directus itself that we should fix 🙂
